src/check/test/type_checking_integration.zig

```diff
@@ -2173,12 +2208,17 @@ test "check type - comprehensive: polymorphism + lambdas + dispatch + annotation
     try checkTypesModule(
         source,
         .{ .pass = .{ .def = "main" } },
+        // TODO: Look at why constraints are not deduped!
```

Comment:
I have another PR that I think will help with this (I forget which one though 😂) - once that one lands I can merge it into this and see if it helps!